security/fma-contract-version-upgrade.md

```diff
@@ -0,0 +1,122 @@
+# [Project Name]: Failure Modes and Recovery Path Analysis
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Introduction](#introduction)
+- [Failure Modes and Recovery Paths](#failure-modes-and-recovery-paths)
+  - [Missing side effect on `.selector` access bug (Bug fixed after 0.8.15)](#missing-side-effect-on-selector-access-bug-bug-fixed-after-0815)
+  - [Storage write removal before conditional termination (Bug fixed after 0.8.15)](#storage-write-removal-before-conditional-termination-bug-fixed-after-0815)
+  - [Change in Default EVM version between different Solidity versions](#change-in-default-evm-version-between-different-solidity-versions)
+- [Audit Requirements](#audit-requirements)
+- [Action Items](#action-items)
+- [Appendix](#appendix)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+|                     |               |
+| ------------------- | ------------- |
+| Author              | Michael Amadi |
+| Created at          | 2024-08-20    |
+| Needs Approval From | Matt Solomon  |
+| Other Reviewers     | Mark Tyneway  |
+| Status              | Draft         |
+
+> [!NOTE]
+> 📢 Remember:
+>
+> - The single approver in the “Need Approval From” must be from the Security team.
+> - Maintain the “Status” property accordingly.
+
+> [!TIP]
+> Guidelines for writing a good analysis, and what the reviewer will look for:
+>
+> - Show your work: Include steps and tools for each conclusion.
+> - Completeness of risks considered.
+> - Include both implementation and operational failure modes
+> - Provide references to support the reviewer.
+> - The size of the document will likely be proportional to the project's complexity.
+> - The ultimate goal of this document is to identify action items to improve the security of the project. The FMA review process can be accelerated by proactively identifying action items during the writing process.
+
+## Introduction
+
+This document covers solidity version upgrade of all OP Stack smart contracts from 0.8.15 code to 0.8.25.
```

Comment:
Per our discord conversation: We might want some of the tstore features in 0.8.27, so I would suggest not assumeing 0.8.25 is the target, and adding an FMA item like "New solidity version is too recent and contains unknown bugs" that discusses this tradeoff